### DISCUSSION TITLE
Add a way to generate a metafile for bundle analsys

### DISCUSSION THREAD

**Discussion by danielo515** [withastro/roadmap#918] [Proposal]
### Body

# Summary

I want to explore why my files are bundled the way they are and see what is being included.

# Background & Motivation

I want to see what is being sent to my FE and why

# Goals

The ideal solution will be a way to say the uderlying esbuild to generate a metafile. But they don't offer such api, you have to get the result from the build and save the metadata included in the result to a JSON file. I tried using the advanced astro build, but that is not working.

# Example

```ts

  vite: {
    build: {
      sourcemap: true,
    },
    optimizeDeps: {
      exclude: ["oslo"],
    },
    define: {
      VERSION: JSON.stringify(version),
    },
    metafile: './metafile.js'
  }
```

**Comment by delucis** [comment]
I haven’t tried these yet myself, but these might might come in handy for this potentially:

- https://www.npmjs.com/package/rollup-plugin-visualizer
- https://www.npmjs.com/package/vite-bundle-analyzer

(Astro is built on Vite, which is built on rollup, so plugins for those underlying tools often work for Astro too)
